default.nix

```diff
@@ -1,21 +1,38 @@
-{ pkgs ? (
-    let
-      inherit (builtins) fetchTree fromJSON readFile;
-      inherit ((fromJSON (readFile ./flake.lock)).nodes) nixpkgs gomod2nix;
-    in
-    import (fetchTree nixpkgs.locked) {
-      overlays = [
-        (import "${fetchTree gomod2nix.locked}/overlay.nix")
-      ];
-    }
-  )
-, buildGoApplication ? pkgs.buildGoApplication
-}:
-
-buildGoApplication {
-  pname = "ess";
-  version = "2.14.1";
-  pwd = ./.;
-  src = ./.;
-  modules = ./gomod2nix.toml;
+{pkgs ? import <nixpkgs> {}}: let
+  build = pkgs.buildGoModule {
+    pname = "ess";
+    version = "2.14.1";
+    src = ./.;
+    pwd = ./.;
+    vendorHash = "sha256-oIA1LTBvkA26ferZI+uOipLE2/wZr2XKjGfESuD/cPk=";
+    meta = {
+      description = "ess (env-sample-sync): automatically and safely synchronize env.sample files with .env";
+      license = pkgs.lib.licenses.bsd2;
+    };
+  };
+in {
+  darwin = build.overrideAttrs (old:
+    old
+    // {
+      GOOS = "darwin";
+      GOARCH = "amd64";
+    });
+  linux32 = build.overrideAttrs (old:
+    old
+    // {
+      GOOS = "linux";
+      GOARCH = "386";
+    });
+  x86_64-linux = build.overrideAttrs (old:
+    old
+    // {
+      GOOS = "linux";
+      GOARCH = "amd64";
+    });
+  windows64 = build.overrideAttrs (old:
+    old
+    // {
+      GOOS = "windows";
+      GOARCH = "amd64";
+    });
```

Comment:
There is a lot of repetition here. You could extract a builder func that takes in the name, goos, goarch, map and build this object.

Reply:
Yeah agreed. My nixlang skills are not good and didn't think I'd get it done cleanly.